src/ORM/DataList.php

```diff
@@ -1321,24 +1376,32 @@ private function fetchEagerLoadManyMany(
      */
     public function eagerLoad(...$relations): static
```

Comment:
This method now stores a list of all partial and complete eagerload relation chains into `eagerLoadAllRelations` to avoid duplicates, and a list of all _complete_ eagerload relation chains into `eagerLoadRelationChains` which `fetchEagerLoadRelations()` iterates over.